src/nft/hooks/usePermit2Approval.ts

```diff
@@ -7,26 +7,28 @@ import usePermit2Allowance, { AllowanceState } from 'hooks/usePermit2Allowance'
 import { useCallback, useMemo, useState } from 'react'
 import invariant from 'tiny-invariant'
 
-// TODO: This should be removed when the sdk is updated to include the new UR address
-const NFT_UNIVERSAL_ROUTER_MAINNET_ADDRESS = '0x4c60051384bd2d3c01bfc845cf5f4b44bcbe9de5'
-
 export default function usePermit2Approval(
-  amount?: CurrencyAmount<Token>,
-  maximumAmount?: CurrencyAmount<Token>,
-  shouldUseNftRouter?: boolean
+  amount: CurrencyAmount<Token> | undefined,
+  maximumAmount: CurrencyAmount<Token> | undefined,
+  nftUniversalRouterContractAddress?: string
 ) {
   const { chainId } = useWeb3React()
 
-  let contractToApprove: string | undefined
-  if (chainId) {
-    const forceNftRouter = shouldUseNftRouter && chainId === 1
-    contractToApprove = forceNftRouter ? NFT_UNIVERSAL_ROUTER_MAINNET_ADDRESS : UNIVERSAL_ROUTER_ADDRESS(chainId)
+  let universalRouterAddress = nftUniversalRouterContractAddress
+  switch (chainId) {
```

Comment:
this switch statement is kinda awkward with the `undefined` case in between - can you just define a helper function to simplify?

```
function getURAddress(chainId?: number, nftURAddress?: string) {
   if (!chainId) return
 
   // if mainnet and on NFT flow, use the contract address returned by GQL
   if (chainId === SupportedChainId.Mainnet) {
       return nftURAddress ?? UNIVERSAL_ROUTER_ADDRESS(chainId)
   }

  return UNIVERSAL_ROUTER_ADDRESS(chainId)
}
```